### PR TITLE
Increase code coverage

### PR DESCRIPTION
**Pull request by TheSTL**
🍰 What's Inside?

- Fix eslint error.
- Wrong file name corrected.
   - src/_DataStructures_/Trees/BinarySearchTree/bst-deletion.test.js → src/_DataStructures_/Trees/BinarySearchTree/height-of-bst/height-of-bst.test.js
  - src/_DataStructures_/Trees/BinarySearchTree/BinarySearchTree.test.js → src/_DataStructures_/Trees/BinarySearchTree/bst-remove.test.js
- Add unit test for BST to increase code coverage.
   - bst-search.test.js
   - bst-minimum.test.js
   - bst-maximum.test.js
   - bst-isEmpty.test.js
- Fix BST remove function. Earlier if we want to remove root and root has only one child then remove function not working correctly.
- Increase code coverage for following algo to near 100 %
      - [TernarySearch](src/_Searching_/TernarySearch/index.js)
      - [JumpSearch](src/_Searching_/JumpSearch/index.js)
      - [get-smallest-common-number](src/_Problems_/get-smallest-common-number/index.js )
      - [anagrams](src/_Problems_/anagrams/anagrams.test.js)
      - [AStar](src/_PathFinder_/AStar/index.js )
      - [lowest-common-ancestor](src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.test.js )
      - height-of-bst
      - [postfix-expression-evaluation]( src/_DataStructures_/Stack/postfix-expression-evaluation/index.js)
      - [Queue]( src/_DataStructures_/Queue/)
      - [LinkedList]( src/_DataStructures_/LinkedList/)
- Add Unit test case for DoublyLinkedList because code coverage is decreased when testing LRUCache testing which is implemented on DoublyLinkedList.
- Create seprate file for Node of DoublyLinkedList

🎉 New Coverage will be  97% after merging.

### CODE REVIEW

File: src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const DLL = require('.');
+
```
Comment (ashokdey) [inline src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js]: Check for DLL should be a Class

File: src/_DataStructures_/Queue/Queue.test.js [hunk 2]
```diff
@@ -53,6 +53,30 @@ describe('Data Structure : Queue', () => {
       expect(queue.dequeue()).toEqual(1);
       expect(queue.dequeue()).toEqual(4);
       expect(queue.dequeue()).toEqual(3);
+      expect(queue.dequeue()).toEqual(null);
+    });
+
+    it('Length of linkedlist', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/Queue/Queue.test.js]: 'Length of linkedlist' should be **'Size of Queue'**

File: src/_DataStructures_/Queue/Queue.test.js [hunk 3]
```diff
@@ -53,6 +53,30 @@ describe('Data Structure : Queue', () => {
...
+      const queue2 = new Queue();
+      queue2.enqueue(2);
+      queue2.enqueue(1);
+      queue2.enqueue(4);
+      queue2.enqueue(3);
+      expect(queue2.length()).toEqual(4);
+    });
+
+    it('Destroy linkedList', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/Queue/Queue.test.js]: 'Destroy linkedList' should be **'Destroy Queue'**

File: src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+describe('Doubly Linked List', () => {
+  const doublyLinkedList = new DLL();
+
+  it('create DLL', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js]: 'create DLL' -> **'It should create a DLL'**

It's all about Behavior Driven Testing, hence it starts with or contains **should**

File: src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    expect(doublyLinkedList.head.next).toEqual(doublyLinkedList.tail);
+    expect(doublyLinkedList.tail.previous).toEqual(doublyLinkedList.head);
+    expect(doublyLinkedList.length()).toEqual(0);
+  });
+
+  it('addAtBeginning', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js]: Change title

File: src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    doublyLinkedList.addAtBeginning(1);
+    doublyLinkedList.addAtBeginning(2);
+    doublyLinkedList.addAtBeginning(3);
+    expect(doublyLinkedList.traverse()).toEqual([3, 2, 1]);
+  });
+
+  it('addAtEnd', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js]: Change title

File: src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    doublyLinkedList.addAtEnd(1);
+    doublyLinkedList.addAtEnd(2);
+    doublyLinkedList.addAtEnd(3);
+    expect(doublyLinkedList.traverse()).toEqual([3, 2, 1, 1, 2, 3]);
+  });
+
+  it('removeAtBeginning', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js]: Change title

File: src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    doublyLinkedList.removeAtBeginning();
+    doublyLinkedList.removeAtBeginning();
+    doublyLinkedList.removeAtBeginning();    
+    expect(doublyLinkedList.traverse()).toEqual([1, 2, 3]);
+  });
+
+  it('removeAtEnd', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/DoublyLinkedList/doublyLinkedList.test.js]: Change title

File: src/_DataStructures_/Trees/BinarySearchTree/bst-maximum.test.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('Binary Search Tree', () => {
+  describe('Find maximum value in BST', () => {
+    const bst = new BinarySearchTree(6);
+    bst.add(4);
+    bst.add(9);
+    bst.add(2);
+    bst.add(5);
+    bst.add(8);
+    bst.add(12);
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-maximum.test.js]: Can we use a loop here?

File: src/_DataStructures_/Trees/BinarySearchTree/bst-minimum.test.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('Binary Search Tree', () => {
+  describe('Find minimum value in BST', () => {
+    const bst = new BinarySearchTree(6);
+    bst.add(4);
+    bst.add(9);
+    bst.add(2);
+    bst.add(5);
+    bst.add(8);
+    bst.add(12);
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-minimum.test.js]: Loop here as well?

File: src/_DataStructures_/Trees/BinarySearchTree/bst-minimum.test.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+const BinarySearchTree = require('./index');
+
+describe('Binary Search Tree', () => {
+  describe('Find minimum value in BST', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-minimum.test.js]: 'Find minimum value in BST' -> **'It should Find the minimum value in BST'**

File: src/_DataStructures_/Trees/BinarySearchTree/bst-minimum.test.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    const bst = new BinarySearchTree(6);
+    bst.add(4);
+    bst.add(9);
+    bst.add(2);
+    bst.add(5);
+    bst.add(8);
+    bst.add(12);
+    it('should return 4', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-minimum.test.js]: Meaningless title...why it should return 4?

File: src/_DataStructures_/Trees/BinarySearchTree/bst-remove.test.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
+const BST = require('.');
+
+describe('Data Structure : Binary Search Tree', () => {
+  it('Should be class', () => {
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-remove.test.js]: 'Should be class' -> **Binary Search Tree should be a Class**

File: src/_DataStructures_/Trees/BinarySearchTree/bst-search.test.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+describe('Binary Search Tree', () => {
+  describe('Find maximum value in BST', () => {
+    const bst = new BinarySearchTree(6);
+
+    bst.add(4);
+    bst.add(9);
+    bst.add(2);
+    bst.add(5);
+    bst.add(8);
+    bst.add(12);
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-search.test.js]: Loop here?

File: src/_DataStructures_/Trees/BinarySearchTree/height-of-bst/height-of-bst.test.js [hunk 15]
```diff
@@ -48,4 +48,20 @@ describe('Binary search tree traversals', () => {
...
+    const bst2 = new BinarySearchTree(10);
+    bst2.add(11);
+    bst2.add(20);
+    bst2.add(9);
+    bst2.add(8);
+    bst2.add(7);
+    bst2.add(6);
+    bst2.add(5);
+    bst2.add(4);
+
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/height-of-bst/height-of-bst.test.js]: Loop here?

File: src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.js [hunk 16]
```diff
@@ -3,18 +3,15 @@
  *
  * Given values of two values n1 and n2 in a Binary Search Tree, find the Lowest Common Ancestor (LCA). You may assume that both the values exist in the tree.
  */
- 
+
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.js]: Write a meaningful function name

File: src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.js [hunk 17]
```diff
@@ -3,18 +3,15 @@
...
 function lca(node, n1, n2) {
-  if (node == null)
-    return null;
+  if (node == null) return null;
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.js]: use **===**

File: src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.test.js [hunk 18]
```diff
@@ -5,8 +5,7 @@ const BinarySearchTree = require('../index');
 // {"left":{"left":{"data":4},"right":{"left":{"data":10},"right":{"data":14},"data":12},"data":8},"right":{"data":22},"data":20}
 
 describe('LCA', () => {
-
-  let bst = new BinarySearchTree(20);
+  const bst = new BinarySearchTree(20);
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/lowest-common-ancestor/index.test.js]: Correct all the titles

File: src/_DataStructures_/Trees/BinarySearchTree/bst-isEmpty.test.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+
+describe('Binary Search Tree', () => {
+  describe('Is Empty', () => {
+    const bst = new BinarySearchTree(6);
+    bst.add(4);
+    bst.add(9);
+    bst.add(2);
+    bst.add(5);
+    bst.add(8);
+    bst.add(12);
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-isEmpty.test.js]: Use loop here

File: src/_DataStructures_/Trees/BinarySearchTree/bst-isEmpty.test.js [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+    });
+
+    it('should return True', () => {
+      bst.remove(6);
+      bst.remove(4);
+      bst.remove(9);
+      bst.remove(2);
+      bst.remove(5);
+      bst.remove(8);
+      bst.remove(12);
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/BinarySearchTree/bst-isEmpty.test.js]: Use loop here

**Comment by TheSTL** [comment]
Thanks @ashokdey . Please review the changes.

**Pull request merged**